src/GitVersion.Core/VersionCalculation/ContinuousDeploymentVersionCalculator.cs

```diff
@@ -0,0 +1,61 @@
+using System.Diagnostics.Contracts;
+using GitVersion.Common;
+using GitVersion.Logging;
+
+namespace GitVersion.VersionCalculation;
+
+internal sealed class ContinuousDeploymentVersionCalculator : NonTrunkBasedVersionCalculatorBase, IContinuousDeploymentVersionCalculator
+{
+    public ContinuousDeploymentVersionCalculator(ILog log, IRepositoryStore repositoryStore, Lazy<GitVersionContext> versionContext)
+        : base(log, repositoryStore, versionContext)
+    {
+    }
+
+    public SemanticVersion Calculate(NextVersion nextVersion)
+    {
+        using (this.log.IndentLog("Using continues deployment typology to calculate the incremented version!!"))
+        {
+            if (!nextVersion.Configuration.IsMainline || nextVersion.Configuration.Label is not null)
+                throw new WarningException("--PRE--CONDITION--FAILED--");
```

Comment:
I'm not entirely sure what the warning message should be here, but please find something more user friendly.

```suggestion
                throw new WarningException("...");
```

Reply:
Done